File: YrradiaSDL2/Game/Cursor/Cursor.cs
namespace YrradiaSDL2.Game.Cursor
{
    public class Cursor
    {
        private const int k_cursorSize = 30;

        public void HideSystemCursor()
        {
            SDL.ShowCursor(0);
        }

        public void Render()
        {
            var mousePosition = _.mouseUtilities.GetMousePosition();
            _.imageRenderer.DrawImage("CursorDefault", mousePosition.X, mousePosition.Y, (int)(k_cursorSize / _.gameProperties.m_resolutionScaleDown), (int)(k_cursorSize / _.gameProperties.m_resolutionScaleDown));
        }
    }
}

File: src/core/cursor/cursor.cpp
/*
 * Copyright 2025 Andreas Åkerberg
 * This code is licensed under MIT license (see LICENSE for details)
 */
#include "cursor.hpp"
#include "core/rendering/images/image_renderer.hpp"

namespace forr {
  void cursor::initialize() { disable_system_cursor(); }

  void cursor::disable_system_cursor() { SDL_ShowCursor(SDL_DISABLE); }

  void cursor::reset_style_to_default() {
    m_cursorStyle = cursor_styles::Default;
  }

  void cursor::render() {
    auto mousePosition{get_normalized_mouse_position()};
    auto width{k_cursorSize};
    auto height{convert_width_to_height(k_cursorSize)};
    str cursorImage;
    switch (m_cursorStyle) {
    case cursor_styles::Default:
      cursorImage = "CursorDefault";
      break;
    case cursor_styles::HoveringClickableGUI:
      cursorImage = "CursorHoveringClickableGUI";
      break;
    }
    get_singleton<image_renderer>().draw_image(
        cursorImage, mousePosition.x - width / 2, mousePosition.y - height / 2,
        width, height);
  }
}